src/test/javascript/spec/component/overview/course-lectures/course-lecture-details.component.spec.ts

```diff
@@ -0,0 +1,214 @@
+            ],
+            providers: [
+                MockProvider(LectureService, {
+                    find: () => {
+                        let headers = new HttpHeaders();
+                        headers = headers.set('Content-Type', 'application/json; charset=utf-8');
+                        return of(new HttpResponse({ body: lecture, headers, status: 200 }));
+                    },
+                }),
+                MockProvider(JhiAlertService),
+                { provide: FileService, useClass: MockFileService },
+                { provide: Router, useValue: mockRouter },
+                { provide: TranslateService, useClass: MockTranslateService },
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        params: of({ lectureId: '1', courseId: '1' }),
+                    },
+                },
+            ],
+            schemas: [],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CourseLectureDetailsComponent);
+                courseLecturesDetailsComponent = fixture.componentInstance;
+                debugElement = fixture.debugElement;
+            });
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    });
+
+    it('should initialize', () => {
+        fixture.detectChanges();
+
+        expect(courseLecturesDetailsComponent).to.be.ok;
+    });
+
+    it('should display all three lecture units: 2 attachment units and 1 text unit', fakeAsync(() => {
+        fixture.detectChanges();
+
+        const attachmentUnits = debugElement.queryAll(By.css('jhi-attachment-unit'));
+        const textUnits = debugElement.queryAll(By.css('jhi-text-unit'));
+        expect(attachmentUnits).to.have.lengthOf(2);
+        expect(textUnits).to.have.lengthOf(1);
+    }));
+
+    it('should display download PDF button', fakeAsync(() => {
+        fixture.detectChanges();
+
+        const downloadButton = debugElement.query(By.css('#downloadButton'));
+        expect(downloadButton).to.exist;
+        expect(courseLecturesDetailsComponent.hasPdfLectureUnit).to.be.true;
+    }));
+
+    it('should not display download PDF button', fakeAsync(() => {
+        lecture.lectureUnits = [lectureUnit3];
+        courseLecturesDetailsComponent.lecture = lecture;
+        courseLecturesDetailsComponent.ngOnInit();
+        fixture.detectChanges();
+
+        const downloadButton = debugElement.query(By.css('#downloadButton'));
+        expect(downloadButton).to.not.exist;
+        expect(courseLecturesDetailsComponent.hasPdfLectureUnit).to.be.false;
+    }));
+
+    it('should download PDF file', fakeAsync(() => {
+        fixture.detectChanges();
+
+        const downloadAttachmentStub = sinon.stub(courseLecturesDetailsComponent, 'downloadMergedFiles');
+        const downloadButton = debugElement.query(By.css('#downloadButton'));
+        expect(downloadButton).to.be.ok;
```

Comment:
Why did you use to.be.ok here instead of to.exist like before?

Reply:
there is no specific reason or at least I don't remember it right not, that's why I will change it to make it consistent with the other tests